pkgs/by-name/cl/cloudflare-ddns/package.nix

```
@@ -0,0 +1,74 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  bash,
+  curl,
+  jq,
+  coreutils,
+  gawk,
+  gnugrep,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cloudflare-ddns";
+  version = "2.0";
+
+  src = fetchFromGitHub {
+    owner = "fernvenue";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-+0AQAa7j6nPaIy0gHEKM1WgPJ7661+NBB1j/dsb7X9Q=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildInputs = [
+    bash
+    curl
+    jq
+    coreutils
+    gawk
+    gnugrep
+  ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp cloudflare-ddns.sh $out/bin/cloudflare-ddns
+    chmod +x $out/bin/cloudflare-ddns
+
+    wrapProgram $out/bin/cloudflare-ddns \
+      --prefix PATH : ${
+        lib.makeBinPath [
+          bash
+          curl
+          jq
+          coreutils
+          gawk
+          gnugrep
+        ]
+      }
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Lightweight Cloudflare DDNS script";
+    longDescription = ''
+      A lightweight script for updating Cloudflare DNS records automatically.
+      Supports IPv4 and IPv6, multiple records, smart monitoring, automatic caching,
+      multiple authentication methods, proxy support, systemd integration,
+      Telegram notifications, CSV logging, and hook commands.
+    '';
+    homepage = "https://github.com/fernvenue/cloudflare-ddns";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ bdim404 ];
+    platforms = platforms.unix;
+    mainProgram = "cloudflare-ddns";
+  };
```

```suggestion
  meta = {
    description = "Lightweight Cloudflare DDNS script";
    longDescription = ''
      A lightweight script for updating Cloudflare DNS records automatically.
      Supports IPv4 and IPv6, multiple records, smart monitoring, automatic caching,
      multiple authentication methods, proxy support, systemd integration,
      Telegram notifications, CSV logging, and hook commands.
    '';
    homepage = "https://github.com/fernvenue/cloudflare-ddns";
    license = lib.licenses.bsd3;
    maintainers = with lib.maintainers; [ bdim404 ];
    platforms = lib.platforms.unix;
    mainProgram = "cloudflare-ddns";
  };
```

Don't use `with lib;`